File: app/src/main/java/top/mikoto/sangnam/Utils/Vibrator.java
package top.mikoto.sangnam.Utils;

import android.content.Context;

public class Vibrator {
    private static Vibrator instance;
    private final android.os.Vibrator vibrator;

    private Vibrator(Context context) {
        Context context1 = context;
        vibrator = (android.os.Vibrator) context1.getSystemService(Context.VIBRATOR_SERVICE);
    }

    public static Vibrator getInstance(Context context) {
        if(instance!=null)
        {
            return instance;
        }
        else
        {
            return instance = new Vibrator(context.getApplicationContext());
        }
    }

    public void vibrate(int millisecond)
    {
        vibrator.vibrate(millisecond);
    }
}

File: app/src/main/java/top/mikoto/sangnam/utils/Vibrator.kt
package top.mikoto.sangnam.utils

import android.content.Context

class Vibrator private constructor(context: Context) {
    private val vibrator: android.os.Vibrator
    fun vibrate(millisecond: Int) {
        vibrator.vibrate(millisecond.toLong())
    }

    companion object {
        private var instance: Vibrator? = null
        fun getInstance(context: Context): Vibrator? {
            return if (instance != null) {
                instance
            } else {
                Vibrator(context.applicationContext).also { instance = it }
            }
        }
    }

    init {
        vibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as android.os.Vibrator
    }
}